internal/venice-common/src/main/java/com/linkedin/venice/kafka/consumer/ApacheKafkaConsumerAdapterFactory.java

```diff
@@ -0,0 +1,42 @@
+package com.linkedin.venice.kafka.consumer;
+
+import com.linkedin.venice.exceptions.VeniceException;
+import com.linkedin.venice.pubsub.api.PubSubConsumerAdapterFactory;
+import com.linkedin.venice.pubsub.api.PubSubMessageDeserializer;
+import com.linkedin.venice.pubsub.consumer.PubSubConsumer;
+import com.linkedin.venice.pubsub.kafka.KafkaPubSubMessageDeserializer;
+import com.linkedin.venice.utils.VeniceProperties;
+import java.io.IOException;
+import org.apache.kafka.clients.consumer.KafkaConsumer;
+
+
+public class ApacheKafkaConsumerAdapterFactory implements PubSubConsumerAdapterFactory<PubSubConsumer> {
+  private static final String NAME = "ApacheKafkaConsumer";
+
+  @Override
+  public ApacheKafkaConsumer create(
+      VeniceProperties veniceProperties,
+      boolean isKafkaConsumerOffsetCollectionEnabled,
+      PubSubMessageDeserializer pubSubMessageDeserializer,
+      String consumerName) {
+    ApacheKafkaConsumerConfig apacheKafkaConsumerConfig = new ApacheKafkaConsumerConfig(veniceProperties, consumerName);
+    if (pubSubMessageDeserializer instanceof KafkaPubSubMessageDeserializer) {
+      return new ApacheKafkaConsumer(
+          new KafkaConsumer<>(apacheKafkaConsumerConfig.getConsumerProperties()),
+          veniceProperties,
+          isKafkaConsumerOffsetCollectionEnabled,
+          (KafkaPubSubMessageDeserializer) pubSubMessageDeserializer);
+    } else {
+      throw new VeniceException("Only support " + KafkaPubSubMessageDeserializer.class);
+    }
+  }
+
+  @Override
+  public String getName() {
+    return NAME;
+  }
+
+  @Override
+  public void close() throws IOException {
```

Comment:
@sushantmane  I think we might not need to let client adapter factory interface (e.g. `PubSubProducerAdapterFactory`) to extends `Closable`? This factory actually has no resources to reclaim, actually.

Reply:
`Close` was added since in some factories we do additional work, for example, shared producer factory.  And we want to clean up leftover resources. 


We can make this method default?